app/controllers/concerns/curation_concerns/curation_concern_controller.rb

```diff
@@ -194,7 +194,17 @@ def curation_concern_from_search_results
       def search_result_document(search_params)
         _, document_list = search_results(search_params)
         raise CanCan::AccessDenied.new(nil, :show) if document_list.empty?
+        if document_suppressed?(document_list)
+          flash[:notice] = I18n.t("curation_concerns.workflow.default.unauthorized")
+          raise CanCan::AccessDenied.new(nil, :show)
+        end
         document_list.first
       end
+
+      def document_suppressed?(document_list)
+        unless current_user
```

Comment:
I'm wondering why we're only suppressing for anonymous users. Shouldn't we suppress for all users who do not have a current workflow role on the document?